backend/src/main/java/com/bakdata/conquery/models/jobs/CalculateCBlocksJob.java

```diff
@@ -63,13 +70,18 @@ public void execute() throws Exception {
 
 				CBlock cBlock = createCBlock(connector, info);
 				cBlock.initIndizes(info.getBucket().getBucketSize());
-				if (treeConcept) {
-					calculateCBlock(cBlock, (ConceptTreeConnector) connector, info);
+				if (connector instanceof ConceptTreeConnector) {
+					final ConceptTreeConnector treeConnector = (ConceptTreeConnector) connector;
+					calculateCBlock(cBlock, treeConnector, info);
 				}
-				else {
+				else if (connector instanceof VirtualConceptConnector) {
 					calculateCBlock(cBlock, (VirtualConceptConnector) connector, info);
 				}
-				setDateRangeIndex(cBlock, info);
+				else {
+					throw new IllegalArgumentException(String.format("Don't know how to handle connectors of type %s", connector.getClass()));
+				}
```

Comment:
Vielleicht ist hier doch ein refactor angebracht in dem man die calculateCBlock funktionalität in den Connector packt. Das entspräche mehr dem OO schema und man würde offen für andere Connectoren-Typen sein